pkg/xcontext/context.go

```diff
@@ -355,7 +371,10 @@ func (ctx *ctxValue) Logger() Logger {
 	if loggerInstance == nil {
 		return nil
 	}
-	ctx.mutationSyncer.Do(ctx.considerPending)
+	ctx.mutationSyncer.Do(func() {
+		ctx.considerPending()
+		loggerInstance = ctx.loadLoggerInstance()
+	})
```

Comment:
Fixing a bug introduced while merging contexts. The problem was that the first run of `Logger()` was returning the logger initialized before `considerPending`, while `considerPending` adds fields/tags to the logger.